### ISSUE TITLE
Fetch correct istioctl binary for OSX operating system

### ISSUE THREAD

**Issue by toffentoffen** [tetratelabs/getmesh#64]
As OSX now support intel and apple silicon ARCHs, we need to fetch the correct istioctl binary.

After installing getmesh and running `getmesh version` i got the following error:
````
getmesh version
getmesh version: 1.1.2
active istioctl: 1.10.3-tetrate-v0
error executing istioctl: fork/exec /Users/marcnavarro/.getmesh/istio/1.10.3-tetrate-v0/bin/istioctl: bad CPU type in executable
````
Seems that although we package a correct getmesh distribution for OSX ARM64, even we generate a correct istioctl binary at [https://dl.getistio.io/public/raw/files/istioctl-1.10.3-tetrate-v0-osx-arm64.tar.gz](https://dl.getistio.io/public/raw/files/istioctl-1.10.3-tetrate-v0-osx-arm64.tar.gz) for OSX ARM64. When getmesh fetches `istioctl` it fetches the binary without arch: 

https://github.com/tetratelabs/getmesh/blob/fd6d94abb71d3eaefc0724558e0636e7f5229b9f/internal/istioctl/istioctl.go#L233-L246

This issue is related to issue #24.

I think that the fix is easy, and I am happy to do it. However I am wondering:
- Are there other places affected by the same issue.
- Is there anything else we need to consider, like for instance updating the getmesh download section for macos to add the option to different archs?

**Comment by mathetake** [comment]
Thanks @toffentoffen, actually looks like the fetchIstoctl function doesn't support ARM64 even though we allow users to install getistio for that architecture (you can see we already build the binary both for osx-arm64 and osx-amd64: 
https://github.com/tetratelabs/getmesh/blob/main/.goreleaser.yml#L8-L13). That means https://github.com/tetratelabs/getmesh/issues/24 is already resolved, but it is not functional at all since it cannot fetch the correct istioctl 😞 

> I think that the fix is easy, and I am happy to do it. However I am wondering:
> Are there other places affected by the same issue.
> Is there anything else we need to consider, like for instance updating the getmesh download section for macos to add the option to different archs?

I would appreciate it you could work on a fix, and no concerns from me! Thank you!

**Comment by toffentoffen** [comment]
Great @mathetake I will work on a fix right away.
Thanks for your help, and feedback will be highly appreciated.

**Comment by toffentoffen** [comment]
Solved by #66
